### DISCUSSION TITLE
Why not use the cli as the projects entrypoint

### DISCUSSION THREAD

**Discussion by WaylonWalker** [kedro-org/kedro#729] [Idea]
If I render out a new project with `kedro==0.17.2` I get the following entrypoint inside of my setup.py.

``` python
entry_point = (
    "new-kedro-project = new_kedro_project.__main__:main"
)
```

If I look at `__main__:main` I see.

``` python
def main():
    configure_project(Path(__file__).parent.name)
    run()
```

Why not utilize the existing projects cli by pointing the entrypoint at `new_kedro_project.cli:cli`?

Just looking for some rationale, or if there would be interest in a PR to change the template.

**Comment by limdauto** [comment]
Hi @WaylonWalker, the rationale is we want to have the same entrypoint for running Kedro in both CLI and package mode, i.e. `python -m new_kedro_project`. This allows us to make sure that both run modes can support the same set of CLI args. To use the `python -m` idiom, however, we need a `__main__.py` as instructed by https://docs.python.org/3/library/__main__.html
